src/Microsoft.AspNetCore.Mvc.Razor/Internal/FileProviderRazorProject.cs

```diff
@@ -13,22 +14,38 @@ public class FileProviderRazorProject : RazorProject
     {
         private const string RazorFileExtension = ".cshtml";
         private readonly IFileProvider _provider;
+        private readonly IHostingEnvironment _hostingEnvironment;
 
-        public FileProviderRazorProject(IRazorViewEngineFileProviderAccessor accessor)
+        public FileProviderRazorProject(IRazorViewEngineFileProviderAccessor accessor, IHostingEnvironment hostingEnviroment)
```

Comment:
This really doesn't seem right to me.

Reply:
Is the problem here that `IFileProvider` doesn't include a way the base path of the root?

Reply:
Yup